tests/unit/multipass/test_installer.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import shutil
+import sys
+from unittest import mock
+
+import pytest
+
+from craft_providers import multipass
+
+
+@pytest.fixture
+def mock_details_from_process_error():
+    details = "<details>"
+    with mock.patch(
+        "craft_providers.multipass.installer.details_from_called_process_error",
+        return_value=details,
+    ) as mock_details:
+        yield mock_details
+
+
+def test_install_darwin(fake_process, monkeypatch):
+    monkeypatch.setattr(sys, "platform", "darwin")
+
+    fake_process.register_subprocess(["brew", "cask", "install", "multipass"])
+    fake_process.register_subprocess(
+        ["multipass", "version"], stdout="multipass 1.4.2\nmultipassd 1.4.2\n"
+    )
+
+    multipass.install()
+
+    assert list(fake_process.calls) == [
+        ["brew", "cask", "install", "multipass"],
+        ["multipass", "version"],
+    ]
+
+
+def test_install_darwin_error(
+    fake_process, mock_details_from_process_error, monkeypatch
+):
+    monkeypatch.setattr(sys, "platform", "darwin")
+    fake_process.register_subprocess(
+        ["brew", "cask", "install", "multipass"],
+        returncode=1,
+    )
+
+    with pytest.raises(multipass.MultipassInstallationError) as exc_info:
+        multipass.install()
+
+    assert exc_info.value == multipass.MultipassInstallationError(
+        reason="error during brew installation",
+        details=mock_details_from_process_error.return_value,
+    )
+
+
+def test_install_linux(fake_process, monkeypatch):
+    monkeypatch.setattr(sys, "platform", "linux")
+
+    fake_process.register_subprocess(["sudo", "snap", "install", "multipass"])
+    fake_process.register_subprocess(
+        ["multipass", "version"], stdout="multipass 1.4.2\nmultipassd 1.4.2\n"
+    )
+
+    multipass.install()
+
+    assert list(fake_process.calls) == [
+        ["sudo", "snap", "install", "multipass"],
+        ["multipass", "version"],
+    ]
+
+
+def test_install_linux_error(
+    fake_process, mock_details_from_process_error, monkeypatch
+):
+    monkeypatch.setattr(sys, "platform", "linux")
+    fake_process.register_subprocess(
+        ["sudo", "snap", "install", "multipass"],
```

Comment:
I wonder if we should just remove sudo here

Reply:
The `snap` command won't install without root?  I expect the application will interact with the user before invoking install().

Reply:
most systems will trigger a snap install over policy kit

Reply:
Ah, you are indeed correct.  I do not have the polkit agent running so it never happened for me before :D  But I started it and it worked as you'd expect.